server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -489,7 +504,13 @@ private MoveDecision explainRebalanceDecision(final ProjectIndex index, final Sh
             assert currentNode != null : "currently assigned node could not be found";
 
             // balance the shard, if a better node can be found
-            final float currentWeight = sorter.getWeightFunction().calculateNodeWeightWithIndex(this, currentNode, index);
+            final float currentWeight = calculateNodeWeightWithIndex(
+                sorter.getWeightFunction(),
+                this,
+                currentNode,
+                index,
+                -Float.MAX_VALUE
+            );
```

Comment:
Do we need special handling here? The above try-catch for `sorter.reset` should guarantee this to be a valid value? If we do need handle it, can it also be try-catch and return `NOT_TAKEN`? Both should help avoid the need for different defaults?